### PR TITLE
Move Phase2 HLT to ticl_v4 

### PR DESCRIPTION
**Pull request by rovere**
#### PR description:

The current Phase2 HLT Menu is a mixture of TICL_v3 and TICL_v4.
This PR will move the whole menu to using TICL_v4. In order to do that, a new option has been added to TICL_v4, during the _tracksters' merge step_, to disable the usage of MTD information that, at present, is not part of the HLT reconstruction.
As an added enhancement, a slimmed and fully L1T-seeded Phase2 HLT menu has been added to the release, named `75e33_timing`, to be used to perform meaningful timing measurements.

For further information, please refer to today's [HLT Upgrade](https://indico.cern.ch/event/1324703/) meeting and presentation therein.

The old TICL_v3 iterations have been removed from the repository, since `GitHub` will automatically keep their history.

A backport to `13_1_X`, the baseline release for the coming HLT Upgrade Annual Review, will be submitted soon.

#### PR validation:

Run any Phase2 workflow with HLT menu w/o crashes.

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by srimanob** [comment]
+Upgrade

Code related to upgrade is fine. Changes can be seen in HLT of Phase-2 workflows which are expected. For Run-3, change in MessageLogger and mkfit workflow do not related to this PR.

File: HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersMerge_cfi.py [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
+
+  ),
+  trackstersclue3d = cms.InputTag('ticlTrackstersCLUE3DHigh'),
+  layer_clusters = cms.InputTag('hgcalMergeLayerClusters'),
+  layer_clustersTime = cms.InputTag('hgcalMergeLayerClusters', 'timeLayerCluster'),
+  tracks = cms.InputTag('generalTracks'),
+  tracksTime = cms.InputTag('tofPID', 't0'),
+  tracksTimeQual = cms.InputTag('mtdTrackQualityMVA', 'mtdQualMVA'),
+  tracksTimeErr = cms.InputTag('tofPID', 'sigmat0'),
+#  muons = cms.InputTag('muons1stStep'),
```
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersMerge_cfi.py:22]: any reason to keep this comment?

File: HLTrigger/Configuration/python/HLT_75e33_timing_cff.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
+import FWCore.ParameterSet.Config as cms
```
[the remark this answers is not in the document]
Comment (rovere) [reply HLTrigger/Configuration/python/HLT_75e33_timing_cff.py:1]: Ciao @mmusich that's the version of the simplified HLT Menu to be used when performing timing measurements.

File: HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersMergeV3_cfi.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import FWCore.ParameterSet.Config as cms
+
+ticlTrackstersMerge = cms.EDProducer("TrackstersMergeProducerV3",
```
Comment (mmusich) [inline HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersMergeV3_cfi.py:3]: I am confused about the usage of this producer. It doesn't seem to be referenced in the menu.
What's the purpose of this configuration fragment?

**Comment by mmusich** [comment]
+hlt
   * [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-12-1100+49c52c/58820/triggerResults): found differences in 2 / 48 workflows, seems to be expected given the  premises https://github.com/cms-sw/cmssw/pull/42759#issue-1892611254
   * there is a couple of minor points that can be clarified https://github.com/cms-sw/cmssw/pull/42759#discussion_r1324071476 and https://github.com/cms-sw/cmssw/pull/42759#discussion_r1326292993
   * given that the PR is coming directly from one of the HLT Upgrade convener, I am implicitly assuming @rovere's blessing

File: HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersMergeV3_cfi.py (hunk 3 above)
Comment (rovere) [reply HLTrigger/Configuration/python/HLT_75e33/modules/ticlTrackstersMergeV3_cfi.py:3]: Ciao @mmusich you are right and that will be cleaned/removed in the (near) future.

**Comment by jfernan2** [comment]
+1

**Comment by antoniovilela** [comment]
+1

**Pull request merged**